backend/app/node.py

```diff
@@ -135,23 +163,78 @@ async def delete_node(
         )
 
     @router.post(
-        "/{tree_id}/{node_id}/edit",
+        "/{tree_id}/{node_id}/{field_set_id}", response_model=NodeValues
     )
     async def edit_node(
         tree_id: str,
         node_id: str,
-        new_props: Dict,
+        field_set_id: str,
+        values: RawNodeValues,
         current_user: UserRead = Depends(fastapi_users.current_user()),
         session: Session = Depends(get_db),
     ):
-        "Editing node"
-        if tree_id not in current_user.trees:
+        "Editing values on node"
+        if (
+            tree_id not in current_user.trees
+            or field_set_id not in current_user.field_set_templates
+        ):
+            raise HTTPException(403)
+
+        tree = await session.Tree.select_id(tree_id)
+        if node_id not in tree.nodes:
+            raise HTTPException(404, "Node not found!")
+
+        node_values = await session.Node.select_related(node_id, NodeValues)
+        relations = list(
+            filter(lambda item: item.out.id == field_set_id, node_values)
+        )
+        if len(relations) > 1:
+            raise HTTPException(500, "Multiple same field_sets for this node!")
+        elif len(relations) == 0:
+            res = await session.NodeValues.create(
+                node_id, field_set_id, **values.dict()
+            )
+        else:
+            relation_id = relations[0].id
+            res = await session.NodeValues.patch(relation_id, **values.dict())
+
+        await session.commit()
+        return res
```

Comment:
I know that is not an issue but can we use response instead of res?

Reply:
I use `res` as in `result`, will do that in the future